src/shogun/mathematics/linalg/GPUMemoryBase.h

```diff
@@ -0,0 +1,19 @@
+#ifndef GPUMemory_Base_H__
+#define GPUMemory_Base_H__
+
+namespace shogun
+{
+
+template <typename T>
+struct GPUMemoryBase
+{
+	GPUMemoryBase()
+	{
+	}
+
+	virtual void transfer_to_CPU(T* data) const = 0;
```

Comment:
we don't need it here as per the design we thought! `to_gpu` and `from_gpu` should be part of the backend.